src/System.Diagnostics.TraceSource/tests/TraceSourceClassTests.cs

```diff
@@ -131,11 +131,20 @@ public void NullSourceName()
         }
 
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "In Desktop in TraceSource.ctor we create the ArgumentException without param name, just with Message = name, so the param name is null.")]
         public void EmptySourceName()
         {
             Assert.Throws<ArgumentException>("name", () => new TraceSource(string.Empty));
             Assert.Throws<ArgumentException>("name", () => new TraceSource(string.Empty, SourceLevels.All));
         }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "In Desktop in TraceSource.ctor we create the ArgumentException without param name, just with Message = name, so the param name is null.")]
+        public void EmptySourceName_Desktop()
+        {
+            Assert.Throws<ArgumentException>(() => new TraceSource(string.Empty));
+            Assert.Throws<ArgumentException>(() => new TraceSource(string.Empty, SourceLevels.All));
```

Comment:
instead of splitting it into 2 methods, you can change the test code to capture the exception and then validate the exception according to the property 

        internal static bool IsFullFramework => RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);